pytorch_pfn_extras/utils/comparer.py

```diff
@@ -475,8 +487,15 @@ def add_engine(self, name, engine, *args, **kwargs):
         if name in self._engines.keys():
             raise ValueError(f"Engine named {name} already registered")
 
+        engine.handler = _ComparableHandler(
+            engine.handler, name, self._compare_targets, self._trigger)
+
+        child_evaluator = getattr(engine, 'evaluator', None)
+        if child_evaluator is not None:
+            # For trainer with evaluator
+            child_evaluator.handler = engine.handler
```

Comment:
Should fail when different handler is set between trainer and evaluator.

Reply:
This fix was already merged in https://github.com/pfnet/pytorch-pfn-extras/pull/425. I will fix it in another PR.

Reply:
For record: #432